launcher/tests/test_updaterapp.py

```diff
@@ -0,0 +1,145 @@
+import os
+import sys
+import pytest
+import subprocess
+from unittest import TestCase
+from unittest import mock
+from unittest.mock import call
+from importlib.machinery import SourceFileLoader
+
+
+# If we're in a PYyQt4 virtual environment, it will be called `.venv-pyqt4` when set up
+# according to our instructions with the pyqt4 venv Makefile target, and we should use
+# the PyQt4 dependencies. Otherwise, use PyQt5
+if sys.prefix.endswith("pyqt4"):
```

Comment:
This is a little bit brittle because it relies on the name of the virtual environment to decide whether we should import PyQt4 or PyQt5. If a user didn't follow the developer docs (which include Makefile targets for PyQt4-based venv and PyQt5 based venv), or if we ever decided to drop those Makefile targets, this check wouldn't work. However, it would default to PyQt5, which is the newer/more common standard. 

I'm open to feedback if there's a better way of managing this for testing purposes though!

Reply:
All prod installs on Qubes 4.0 are still using PyQt4, via F25 in dom0, so that should be the default. The logic in the `if` statement on L14 seems to invert the "use 4 unless 5 is requested" pragma introduced back in #610. Taking a closer look, we're using Python3.8 in CI to test the launcher code, but dom0 is still on 3.5. This isn't a regression introduced by your changes, @rocodes, but your changes are surfacing them. =) I'll take a look at what switching CI over to a more prod-like testing environment would take, and report back.

Reply:
Thanks for fixing CI, lifesaver. :) 

My comment here (around PyQt4 vs PyQt5 being the default) was only relevant to machines that are performing unit testing (for machines that run `test_updaterapp.py` specifically).  But in any case, I've reworked this section to flip the logic to be more consistent with the rest of the codebase, and to check the `SDW_UPDATER_QT` environment variable to decide which version of PyQt to import (note: this means test environments must follow the dev docs and always set the environment variable before running tests).